src/LmdbCache.ts

```diff
@@ -216,4 +254,63 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
       sizeAfter: statsAfter.mapSize
     };
   }
+
+  async begin(): Promise<void> {
+    await this.checkPreviousTransactionFinished();
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    await this.db.put(this.ongoingTransactionMark, 'ongoing');
+    this.rollbackBatch = () => {
+      this.db.removeSync(this.ongoingTransactionMark);
+    };
+    return;
+  }
+
+  private async checkPreviousTransactionFinished() {
+    const transactionMarkValue = (await this.db.get(this.ongoingTransactionMark)) as unknown as string;
+
+    if (transactionMarkValue == 'ongoing') {
+      throw new Error(`Database seems to be in inconsistent state. The previous transaction has not finished.`);
+    }
+  }
+
+  commit(): void {
+    this.db.removeSync(this.ongoingTransactionMark);
+  }
+
+  async kvMap(sortKey: string, options?: SortKeyCacheRangeOptions): Promise<Map<string, V>> {
+    const rangeOptions: RangeOptions = {
+      start: options?.reverse ? options?.lt : options?.gte,
+      end: options?.reverse ? options?.gte : options?.lt,
+      reverse: options?.reverse
+    };
+
+    const result: Map<string, V> = new Map();
+    const rangedKeys = this.db.getKeys(rangeOptions).filter((k) => k != this.ongoingTransactionMark);
+    for (const joinedKey of rangedKeys) {
+      const clientKey = joinedKey.split(this.subLevelSeparator)[0];
+      const wrappedValue = await this.getLessOrEqual(clientKey, sortKey);
+      if (wrappedValue) {
+        result.set(clientKey, (await this.getLessOrEqual(clientKey, sortKey)).cachedValue);
+      }
+    }
+
+    if (options?.limit) {
+      const limitedResult: Map<string, V> = new Map();
+      for (const item of Array.from(result.entries()).slice(0, options.limit)) {
+        limitedResult.set(item[0], item[1]);
+      }
+      return limitedResult;
+    }
+
+    return result;
+  }
+
+  async rollback(): Promise<void> {
+    await this.db.transactionSync(this.rollbackBatch);
+  }
+}
+
+class ClientValueWrapper<V> {
```

Comment:
shouldn't this be imported from the SDK?

Reply:
Well..., maybe, i dont know...
ClientValueWrapper is a helper class to implement delete functionality in leveldb. 
It is very much related to how we use leveldb, but I don't know if we should export it as part of the api.

It so happens that lmdb also implements delete in the similar way, so this class looks the same, but it doesn't have to by like that. 
Maybe the ClientValueWrapper for leveldb will change in a different way then the one for the lmdb.

Reply:
Ok.